src/permutation.jl

```diff
@@ -0,0 +1,65 @@
+export PermutationTest
+
+function ptstats(x,y,f)
+    xy = vcat(x,y)
+    rx = 1 : length(x)
+    ry = length(xy)-length(y)+1 : length(xy)
+    T = typeof(f(xy))  # is there a better way?  e.g., does a funtype() exist, like eltype()??
+    (xy, rx, ry, T)
+end
+
+immutable PermutationTest <: HypothesisTest
+    observation
+    samples
+end
+
+"""
+    PermutationTest(x::Vector, y::Vector, f::Function [, n::Int])
+
+Perform a permutation test (a.k.a. randomization test) of the null hypothesis
+that the probability of `f(x)` is equal to `f(y)`.
```

Comment:
Is is really the "probability"? IIUC that could be any value, right?

Reply:
when pvalue is called on a PermutationTest it returns a value between 0 and 1.  what would you prefer to call it other than a probability?

Reply:
If you refer to the p-value, the wording is wrong. Shouldn't it be something like "of the null hypothesis that `f(x)` is equal to `f(y)`"? Then the p-value is something different which can be obtained via `pvalue` and which can be mentioned later, but is separate from the definition of the tested hypothesis. See examples in other tests.